### PR TITLE
feat: add Asai_file

### LINKED ISSUE #14: :bulb: Intermediate representation
```ocaml
open Bwd

(** Styles *)
type style = [`Context | `Highlight | `Mark]

(** A segment is a styled string without control characters. *)
type marked_string = style * string

(** A line is a list of segments. *)
type marked_line = marked_string list

(** A block is a collection of consecutive lines. *)
type marked_block = { start_line_num : int; text : marked_line list }

(** A file consists of multiple blocks. *)
type marked_file = { file_path : string; blocks : marked_block list }

(** A multi-span consists of all formatted spans across multiple files. *)
type marked_message = marked_file list * Asai.Diagnostic.message

(** a message *)
type t =
  { code : string
  ; severity : Asai.Severity.t
  ; message : marked_message
  ; traces : marked_file bwd
  }
```

### PR DESCRIPTION
**Pull request by favonia**
Implement the compiler. Close #14.

### CODE REVIEW

**Comment by favonia** [comment]
Blocked by #16.

**Comment by favonia** [comment]
I can update this PR after #17.

**Review by TOTBWF — CHANGES_REQUESTED** [review]

File: src/file/Marker.ml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+module R = Reader
+
+type position = Asai.Span.position
+
+let find_eol (pos : Asai.Span.position) =
+  let eof = R.length pos.file_path in
+  let rec go i =
+    if i < eof && R.unsafe_get pos.file_path i = '\n' then
```
[the remark this answers is not in the document]
Comment (favonia) [reply src/file/Marker.ml]: Fixed.

File: src/file/Asai_file.ml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+open Bwd
+
+module Marked = Marked
+
+module type S =
+sig
+  module Code : Asai.Code.S
+  val format : splitting_threshold:int -> Code.t Asai.Diagnostic.t -> Code.t Marked.t
```
[the remark this answers is not in the document]
Comment (favonia) [reply src/file/Asai_file.ml]: Update: use `assemble`

**Pull request merged**